acceptance/api/v1/application_batch_delete_test.go

```diff
@@ -0,0 +1,106 @@
+package v1_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/epinio/epinio/acceptance/helpers/catalog"
+	api "github.com/epinio/epinio/internal/api/v1"
+	"github.com/epinio/epinio/pkg/api/core/v1/models"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AppBatchDelete Endpoint", func() {
+	var (
+		namespace, app1, app2 string
+	)
+	containerImageURL := "splatform/sample-app"
+
+	BeforeEach(func() {
+		namespace = catalog.NewNamespaceName()
+		env.SetupAndTargetNamespace(namespace)
+
+		for _, a := range []*string{&app1, &app2} {
+			*a = catalog.NewAppName()
+			env.MakeContainerImageApp(*a, 1, containerImageURL)
+
+			configurationName := *a + "-conf"
+			env.MakeConfiguration(configurationName)
+			env.BindAppConfiguration(app1, configurationName, namespace)
+		}
+	})
+
+	AfterEach(func() {
+		env.DeleteNamespace(namespace)
+	})
+
+	It("removes the applications and unbinds configurations", func() {
+		responseBody := makeApplicationDeleteRequest(namespace, "", app1, app2)
+		validateApplicationDeletionResponse(responseBody, app1+"-conf", app2+"-conf")
+		verifyApplicationsDeleted(namespace, app1, app2)
+	})
+})
+
+func makeApplicationDeleteRequest(namespace, requestBody string, applicationNames ...string) []byte {
+	q := url.Values{}
+	for _, a := range applicationNames {
+		q.Add("applications[]", a)
+	}
+	URLParams := q.Encode()
+
+	response, err := env.Curl("DELETE", fmt.Sprintf("%s%s/namespaces/%s/applications?%s",
+		serverURL, api.Root, namespace, URLParams), strings.NewReader(requestBody))
+	Expect(err).ToNot(HaveOccurred())
+	Expect(response).ToNot(BeNil())
+
+	defer response.Body.Close()
+	bodyBytes, err := io.ReadAll(response.Body)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(response.StatusCode).To(Equal(http.StatusOK), string(bodyBytes))
+
+	var responseData models.Response
+	err = json.Unmarshal(bodyBytes, &responseData)
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
Shouldn't we parse an `ApplicationDeleteResponse` and check the parsed resp?

Reply:
Actually no, I just need to put the bodyBytes (as a string) in the expectation, to be able to tell what went wrong. No need to parse the response, since we return the response to be parsed in the `verifyApplicationsDeleted` function.

Reply:
Yes, I was actually seeing that we unmarshal again the response as a map in the `validateApplicationDeletionResponse` though. Should we use the `ApplicationDeleteResponse` there at least, instead of a generic map? Or we usually do this kind of "soft" checks?

Reply:
I switched to `ApplicationDeleteResponse` in the `validateApplicationDeletionResponse` function